crates/uv-requirements/src/pyproject.rs

```diff
@@ -478,10 +492,16 @@ pub(crate) fn lower_requirement(
             path_source(path, project_dir, editable)?
         }
         Source::Registry { index } => match requirement.version_or_url {
-            None => RequirementSource::Registry {
-                specifier: VersionSpecifiers::empty(),
-                index: Some(index),
-            },
+            None => {
+                warn_user!(
+                    "You did not specify a version constraint (e.g. a lower bound) for {}",
```

Comment:
Can we frame this without the "You" i.e. "No version constraint (e.g. ...) specified for {}"